src/vs/workbench/contrib/chat/browser/actions/chatContextActions.ts

```diff
@@ -508,22 +512,27 @@ export class AttachContextAction extends Action2 {
 
 			qp.items = items;
 			qp.busy = false;
+		} else if (isAsyncIterable(pickerConfig.picks)) {
+			for await (const items of cancellableIterable(pickerConfig.picks, cts.token)) {
+				qp.items = ([] as QuickPickItem[]).concat(items, extraPicks);
```

Comment:
Doesn't this loose the previous chunks of `items`?

Reply:
Yea, I document that on the docblock for `pickerConfig.picks`. For MCP servers I am requesting all resources from servers in parallel (which can paginate) so I need to replace the entire list in the event that I read a new page of data for a server in the middle of the list.

It is a little funny use of iterators, although I document it in the docblock on the `picks` property. Although thought about using `IProgress` but there's not a good way to get that passed in atm

Reply:
Hm, can you compute a delta in MCP land so that we can have normal iterator semantics? Otherwise folks that use the async-iter in the future will rightfully be confused

Reply:
would the progress/callback logic then be better? like pass `IProgress<T>` to it?

Reply:
Yea, I thought about that, but we'll need to massage these types some more since the function signature for `picks` switches to query-based approach which is not what we want to do here. I can do that though.

Reply:
I opted to go for an Observable approach which also simplifies the logic in the AttachContextAction. Figure we can let consumers do the right thing and it also works in this case without exploding the types